city_scrapers/spiders/det_employment_solutions_corp.py

```diff
@@ -0,0 +1,156 @@
+
+    def _parse_schedule(self, response):
+        """Parse PDF and then yield to documents page"""
+        self._parse_schedule_pdf(response)
+        yield scrapy.Request(
+            response.urljoin(self.docs_link), callback=self._parse_documents, dont_filter=True
+        )
+
+    def _parse_schedule_pdf(self, response):
+        """Parse dates and details from schedule PDF"""
+        pdf_obj = PdfFileReader(BytesIO(response.body))
+        # need to get all pages of pdf schedule
+        number_of_pages = pdf_obj.getNumPages()
+        pdf_text = ""
+        for page_number in range(number_of_pages):
+            pdf_text += pdf_obj.getPage(page_number).extractText()
+
+        # Remove duplicate characters split onto separate lines
+        clean_text = re.sub(r"([A-Z0-9:\n ]{2})\1", r"\1", pdf_text, flags=re.M)
+        # Join lines where there's only a single character, then remove newlines
+        clean_text = re.sub(r"(?<=[A-Z0-9:])\n", "", clean_text, flags=re.M).replace("\n", " ")
+        # Remove duplicate spaces
+        clean_text = re.sub(r"\s+", " ", clean_text)
+        # FIXME must delete later
+        print(clean_text)
+        year_str = re.search(r"\d{4}", clean_text).group()
+        self._validate_location(clean_text)
+
+        time_strs = re.findall(
+            r"\d{1,2}[:]\d{2}\s*[a|p|A|P][m|M].*?\d{1,2}[:]\d{2}\s*[a|p|A|P][m|M]", clean_text)
+
+        print(time_strs)
+        time_tuples = []
+        for time_str in time_strs:
+            time_tuples.append(tuple(re.findall(r"\d{1,2}[:]\d{2}\s*[a|p|A|P][m|M]", time_str)))
+        # extra space to deal with situations like J uly (which is present in test case)
+        date_strs = re.findall(
+            r"[J|F|M|A|S|O|N|D][a-z|]{0,8}\s?[a-z]{1,8}\s+\d{1,2}(?!\d)", clean_text)
+
+        if len(time_tuples) != len(date_strs):
+            raise ValueError("Not all dates can be matched with start and end time")
+
+        for i in range(len(time_strs)):
+            date_str = date_strs[i]
+            try:
+                datetime.strptime("{} {} {}".format(
+                    date_str, year_str, '10:30 am'), "%B %d %Y %I:%M %p")
+            except ValueError:
+                date_str = date_str.replace(" ", "", 1)
+            time_str = time_tuples[i]
+            start_time = time_str[0]
+            end_time = time_str[1]
+            self.meeting_dates.append({'start': self._parse_datetime(
+                start_time, date_str, year_str), 'end':
+                self._parse_datetime(
+                    end_time, date_str, year_str)})
+
+    def _parse_documents(self, response):
+        """Parse agenda and minutes page"""
+        link_map = self._parse_link_map(response)
+        for date in self.meeting_dates:
+            meeting = Meeting(
+                # FIXME may be able to be more specific by by matching
+                # with links found with _parse_link_map?
+                title="Detroit Employment Solutions Corporation",
```

Comment:
This might be difficult, but we'd want to differentiate between meetings of the Board and different committees

Reply:
Okay, I will submit a pr when I have that (and the links) working